astropy/coordinates/erfa_astrom.py

```diff
@@ -0,0 +1,260 @@
+        )
+        earth_pv, earth_heliocentric = prepare_earth_position_vel(frame.obstime)
+        return erfa.apcs(jd1_tt, jd2_tt, pv, earth_pv, earth_heliocentric)
+
+    @staticmethod
+    def apio13(frame):
+        '''
+        Wrapper for ``erfa.apio13``, used in conversions AltAz <-> CIRS
+
+        Arguments
+        ---------
+        frame: ``astropy.coordinates.AltAz``
+        '''
+        lon, lat, height = frame.location.to_geodetic('WGS84')
+
+        jd1_utc, jd2_utc = get_jd12(frame.obstime, 'utc')
+        dut1utc = get_dut1utc(frame.obstime)
+
+        return erfa.apio13(
+            jd1_utc, jd2_utc, dut1utc,
+            lon.to_value(u.radian),
+            lat.to_value(u.radian),
+            height.to_value(u.m),
+            *get_polar_motion(frame.obstime),
+            # all below are already in correct units because they are QuantityFrameAttribues
+            frame.pressure.value,
+            frame.temperature.value,
+            frame.relative_humidity.value,
+            frame.obswl.value,
+        )
+
+
+class ErfaAstromInterpolator(ErfaAstrom):
+    '''
+    A provider for astrometry values that does not call erfa
+    for each individual timestamp but interpolates linearly
+    between support points.
+
+    For the interpolation, float64 MJD values are used, so time precision
+    for the interpolation will be around a microsecond.
+
+    This can dramatically speed up coordinate transformations,
+    e.g. between CIRS and ICRS,
+    when obstime is an array of many values (factors of 10 to > 100 depending
+    on the selected resolution, number of points and the time range of the values).
+
+    The precision of the transformation will still be in the order of microseconds
+    for reasonable values of time_resolution, e.g. ``300 * u.s``.
+
+    Users should benchmark performance and accuracy with the default transformation
+    for their specific use case and then choose a suitable ``time_resolution``
+    from there.
+
+    This class is intended be used together with the ``erfa_astrom`` science state,
+    e.g. in a context manager like this
+
+    Example
+    -------
+    >>> from astropy.coordinates import SkyCoord, CIRS
+    >>> from astropy.coordinates.erfa_astrom import erfa_astrom, ErfaAstromInterpolator
+    >>> import astropy.units as u
+    >>> from astropy.time import Time
+    >>> import numpy as np
+
+    >>> obstime = Time.now() + np.linspace(0, 4, 1000) * u.hour
+    >>> with erfa_astrom.set(ErfaAstromInterpolator(300 * u.s)):
+    ...    cirs = SkyCoord.from_name('Crab').transform_to(CIRS(obstime=obstime))  # doctest: +REMOTE_DATA
+    '''
+
+    @u.quantity_input(time_resolution=u.day)
+    def __init__(self, time_resolution):
+        if time_resolution.to_value(u.ms) < 1:
+            warnings.warn(
+                f'Using {self.__class__.__name__} with `time_resolution`'
+                ' below 1ms might not improve performance',
```

Comment:
I'm not sure, if this warning should really be raised. First of all, the value of 1 ms seems arbitrary to me. Whether there is a speed-up depends entirely on how often the ERFA routines must be queried, but this is dependent on the number of support points, which itself depend on the ratio of the "spread" of the times and the resolution. If all observations were done within few milliseconds, then a resolution of ms will still lead to a speed-up. Whether it is necessary, is a different questions and depends on the required accuracy. So perhaps it would be better to move this discussion into the docs, and add a table or a plot, which demonstrates the achievable accuracy as a function of resolution?

Reply:
Agreed.